### PR TITLE
Update dependecies

### PR DESCRIPTION
**Pull request by janisz**

### CODE REVIEW

**Comment by klueska** [comment]
Is this basically just a move to use `dep`?  If so, we should add a Make target to update the deps.

**Comment by janisz** [comment]
Dep was used but not all information was provided. I updated dependencies and dep configuration files.

**Comment by klueska** [comment]
Do we know why this is failing CI? Is it due to using a different version of Go that doesn't support the `-L` flag?

**Comment by janisz** [comment]
Yep, I'm fixing Dockerfile.

**Comment by janisz** [comment]
@klueska Fixed

File: dcos-log/api/v2/handlers.go [hunk 1]
```diff
@@ -637,7 +637,7 @@ func browseFiles(w http.ResponseWriter, req *http.Request) {
 	}
 
 	if err := json.NewEncoder(w).Encode(files); err != nil {
-		logError(w, req, fmt.Sprintf("unable to encode sandbox files: %s. Items: %s", err, files), http.StatusInternalServerError)
+		logError(w, req, fmt.Sprintf("unable to encode sandbox files: %s. Items: %x", err, files), http.StatusInternalServerError)
```
Comment (klueska) [inline dcos-log/api/v2/handlers.go:640]: This seems unrelated to the message included in this commit message "Update dependecies". It should probably be in its own commit.

File: Makefile [hunk 2]
```diff
@@ -1,45 +1,59 @@
...
-	$(MAKE) -C $* install
-
-build: $(SUBBUILD)
-
-$(SUBBUILD): %.build:
-	$(MAKE) -C $* build
-
-.PHONY: all $(SUBDIRS) clean $(SUBCLEAN) test $(SUBTEST) lint $(SUBLINT) \
-	install $(SUBINSTALL) build $(SUBBUILD)
+IMAGE_NAME=dcos-log-dev
```
Comment (klueska) [inline Makefile:1]: I understand that changes to the Makefile are necessary when removing the extra `dcos-log` directory, but alot has changed in this file that is unrelated to this change. It would be better to make the change incrementally (with separate commits) so that I can easily review this and be *sure* that it does what's expected.

File: api/middleware/download.go [hunk 3]
```diff
@@ -7,6 +7,7 @@ import (
 	"strings"
 
 	"compress/gzip"
+
```
Comment (klueska) [inline api/middleware/download.go:10]: Seems unrelated to this commit

File: api/server.go [hunk 4]
```diff
@@ -8,12 +8,12 @@ import (
...
 	"time"
 
-	"github.com/sirupsen/logrus"
 	"github.com/coreos/go-systemd/activation"
 	"github.com/dcos/dcos-go/dcos"
 	"github.com/dcos/dcos-go/dcos/http/transport"
 	"github.com/dcos/dcos-go/dcos/nodeutil"
-	"github.com/dcos/dcos-log/dcos-log/config"
+	"github.com/dcos/dcos-log/config"
+	"github.com/sirupsen/logrus"
```
Comment (klueska) [inline api/server.go:16]: Seems unrelated to this commit

File: api/v1/handlers.go [hunk 5]
```diff
@@ -11,9 +11,9 @@ import (
 	"strings"
 	"time"
 
-	"github.com/sirupsen/logrus"
-	"github.com/dcos/dcos-log/dcos-log/journal/reader"
+	"github.com/dcos/dcos-log/journal/reader"
 	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
```
Comment (klueska) [inline api/v1/handlers.go:16]: Seems unrelated to this commit

File: api/v1/handlers.go [hunk 6]
```diff
@@ -290,7 +290,7 @@ func readJournalHandler(w http.ResponseWriter, req *http.Request) {
 				return
 			}
 		case <-time.After(time.Second):
-			err := j.Follow(time.Millisecond * 100, w)
+			err := j.Follow(time.Millisecond*100, w)
```
Comment (klueska) [inline api/v1/handlers.go:293]: Seems unrelated to this commit

File: api/v2/handlers.go [hunk 7]
```diff
@@ -594,8 +594,8 @@ func journalHandler(w http.ResponseWriter, req *http.Request) {
 				logrus.Debugf("closing a client connection.")
 				return
 			}
-		case <- time.After(time.Second):
-			err := j.Follow(time.Millisecond * 100, w)
+		case <-time.After(time.Second):
```
Comment (klueska) [inline api/v2/handlers.go:597]: Seems unrelated to this commit

File: journal/reader/read_test.go [hunk 8]
```diff
@@ -8,10 +8,11 @@ import (
 	"testing"
 	"time"
 
-	"github.com/coreos/go-systemd/journal"
 	"bytes"
 	"context"
 	"io"
+
+	"github.com/coreos/go-systemd/journal"
```
Comment (klueska) [inline journal/reader/read_test.go:15]: Seems unrelated to this commit

File: journal/reader/read_test.go [hunk 9]
```diff
@@ -230,7 +231,7 @@ func TestFollow(t *testing.T) {
 	messageCounter := 0
 	for {
 		select {
-		case <- ctx.Done():
+		case <-ctx.Done():
```
Comment (klueska) [inline journal/reader/read_test.go:234]: Seems unrelated to this commit

File: mesos/files/reader/read_test.go [hunk 10]
```diff
@@ -288,4 +288,4 @@ func TestSkipBoundary(t *testing.T) {
 	for i := -100; i < 100; i++ {
 		doRead(t, data, OptReadDirection(BottomToTop), OptSkip(i))
 	}
-}
\ No newline at end of file
+}
```
Comment (klueska) [inline mesos/files/reader/read_test.go:291]: Seems unrelated to this commit

File: Gopkg.toml [hunk 11]
```diff
@@ -1,6 +1,6 @@
 # Gopkg.toml example
 #
-# Refer to https://github.com/golang/dep/blob/master/docs/Gopkg.toml.md
+# Refer to https://golang.github.io/dep/docs/Gopkg.toml.html
```
Comment (klueska) [inline Gopkg.toml:3]: Seems unrelated to this commit

File: Gopkg.toml [hunk 12]
```diff
@@ -16,10 +16,36 @@
...
-#  version = "2.4.0"
+#   name = "github.com/x/y"
+#   version = "2.4.0"
+#
+# [prune]
+#   non-go = false
+#   go-tests = true
+#   unused-packages = true
+
+required = ["golang.org/x/lint/golint", "github.com/golang/lint/golint"]
```
Comment (klueska) [inline Gopkg.toml]: This change seems related, but the ones below seems unrelated to this commit

File: api/server.go [hunk 13]
```diff
@@ -40,7 +40,7 @@ func newNodeInfo(cfg *config.Config, client *http.Client) (nodeutil.NodeInfo, er
 
 // StartServer is an entry point to dcos-log service.
 func StartServer(cfg *config.Config) error {
-	transportOptions := []transport.OptionTransportFunc{}
+	var transportOptions []transport.OptionTransportFunc
```
Comment (klueska) [inline api/server.go:43]: Seems unrelated to this commit

File: api/server.go [hunk 14]
```diff
@@ -72,7 +72,7 @@ func StartServer(cfg *config.Config) error {
 		return err
 	}
 
-	listeners, err := activation.Listeners(true)
+	listeners, err := activation.Listeners()
```
Comment (klueska) [inline api/server.go:75]: Seems unrelated to this commit

File: Dockerfile [hunk 15]
```diff
@@ -1,8 +1,7 @@
 FROM golang:1.11
 
-ENV PATH /go/bin:/usr/local/go/bin:$PATH
-ENV GOPATH /go
-
 RUN apt-get update && apt-get install -y \
     libsystemd-dev \
     init
+
+RUN go get -u golang.org/x/lint/golint
```
Comment (klueska) [inline Dockerfile:7]: Didn't you just remove this in the last commit? We should clean up the commit history here a bit so that it is more easily reviewable.

**Review by klueska — COMMENTED** [review]
Overall, I _think_ I agree with all the changes here, but the way the PR is put together makes it hard to review and verify.

**Comment by janisz** [comment]
Closing in favour of https://github.com/dcos/dcos-log/pull/65

File: api/server.go (hunk 14 above)
Comment (janisz) [reply api/server.go:75]: After updating dependencies `activation.isteners` changed its signature and does not allow parameters.

File: Makefile (hunk 2 above)
Comment (janisz) [reply Makefile:1]: This is the Makefile from subdirectory we need only one Makefile

File: dcos-log/api/v2/handlers.go (hunk 1 above)
Comment (janisz) [reply dcos-log/api/v2/handlers.go:640]: Latest `go vet` catch this errors and build will fail.
